pkg/kubelet/kubelet_pods.go

```diff
@@ -887,6 +888,32 @@ func (kl *Kubelet) podAndContainersAreTerminal(pod *v1.Pod) (containersTerminal,
 	return
 }
 
+// check if terminated pod is kept for more than PodTerminatedStatePeriod.
+// Intend to keep the terminated pod around so that infra container is kept
+// alive while devices are release from it.
+func (kl *Kubelet) podIsInTerminatedStateBeyondGracePeriod(pod *v1.Pod) bool {
+	if !kl.podIsTerminated(pod) {
+		return false
+	}
+
+	var finishTime  time.Time
+	for _, status := range pod.Status.ContainerStatuses {
+		if finishTime.IsZero() {
+			finishTime = status.State.Terminated.FinishedAt.Time
+			continue
+		}
+		if finishTime.After(status.State.Terminated.FinishedAt.Time) {
+			continue
+		}
+		finishTime = status.State.Terminated.FinishedAt.Time
+	}
+	terminatedStateDuration := time.Now().Sub(finishTime)
```

Comment:
Need to look into code, but there is a function to get time.Now in kubelet code. We should use that because time.Now() will add the monotonic clock to it. And Sub() function on it will give you wrong results.